repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/lineage/LineagePreProcessor.java

```diff
@@ -0,0 +1,424 @@
+
+            checkAndUpdateConnectionLineage(inputConnection, true);
+            checkAndUpdateConnectionLineage(outputConnection, true);
+        } finally {
+            RequestContext.get().setSkipAuthorizationCheck(false);
+        }
+
+        return processEntity;
+    }
+
+    @Override
+    public void processDelete(AtlasVertex vertex) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("processDeleteLineageProcess");
+
+        try {
+            Set<String> connectionProcessQNs = getConnectionProcessQNs(vertex);
+            if (connectionProcessQNs.isEmpty()) {
+                return;
+            }
+
+            for (String connectionProcessQn : connectionProcessQNs) {
+                if (!checkIfChildProcessExistForConnectionProcess(connectionProcessQn)) {
+                    deleteConnectionProcess(connectionProcessQn);
+                }
+            }
+        }
+        catch (Exception exp) {
+            LOG.error("Error in LineagePreProcessor.processDelete", exp);
+        }
+        finally {
+            RequestContext.get().endMetricRecord(metricRecorder);
+        }
+    }
+
+    private boolean checkIfChildProcessExistForConnectionProcess(String connectionProcessQn) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("checkIfMoreChildProcessExistForConnectionProcess");
+
+        try {
+            List<Map<String, Object>> mustClauseList = Arrays.asList(
+                    mapOf("term", mapOf("__typeName.keyword", PROCESS_ENTITY_TYPE)),
+                    mapOf("term", mapOf("__state", "ACTIVE")),
+                    mapOf("term", mapOf(PARENT_CONNECTION_PROCESS_QUALIFIED_NAME, connectionProcessQn))
+            );
+
+            Map<String, Object> dsl = mapOf("query", mapOf("bool", mapOf("must", mustClauseList)));
+
+            List<AtlasEntityHeader> process = indexSearchPaginated(dsl, new HashSet<>(Arrays.asList(PARENT_CONNECTION_PROCESS_QUALIFIED_NAME)), this.discovery);
+
+            return CollectionUtils.isNotEmpty(process) && process.size() > 1;
+        } finally {
+            RequestContext.get().endMetricRecord(metricRecorder);
+        }
+    }
+
+    private Set<String> getConnectionProcessQNs(AtlasVertex vertex) {
+        Set<String> connectionProcessQNs = new HashSet<>();
+        try {
+            Iterable<Object> values = vertex.getPropertyValues(PARENT_CONNECTION_PROCESS_QUALIFIED_NAME, Object.class);
+            if (values != null) {
+                for (Object value : values) {
+                    if (value != null) {
+                        connectionProcessQNs.add(value.toString());
+                    }
+                }
+            }
+        } catch (Exception e) {
+            try {
+                String value = vertex.getProperty(PARENT_CONNECTION_PROCESS_QUALIFIED_NAME, String.class);
```

Comment:
Again, why are you expecting String PARENT_CONNECTION_PROCESS_QUALIFIED_NAME? I think this is not needed